spec/helpers/application_helper_spec.rb

```diff
@@ -364,4 +364,50 @@
       end
     end
   end
+
+  describe "#commenter_id_for_css_classes" do
+    context "when commenter is a user" do
+      let(:user) { create(:user) }
+      let(:comment) { create(:comment, pseud: user.default_pseud) }
+
+      it "returns commenter id css class name" do
+        expect(helper.commenter_id_for_css_classes(comment)).to eq("user-#{user.id}")
+      end
+    end
+    
+    context "when commenter is a visitor" do
+      let(:comment) { create(:comment, :by_guest) }
+
+      it "returns nil" do
+        expect(helper.commenter_id_for_css_classes(comment)).to eq(nil)
+      end
+    end
+
+    context "when commenter is creater of work inside anonymous collection" do
+      let(:user) { create(:user) }
+      let(:anonymous_collection) { create(:anonymous_collection) }
+      let(:work) { create(:work, authors: [user.default_pseud]) }
```

Comment:
The code for work anonymity is very finicky at the moment. I'm not 100% certain, but I'm guessing that it'd help to change the definition here to:
```
      let(:work) { create(:work, authors: [user.default_pseud], collections: [anonymous_collection]) }
```
The other alternative would be calling `work.reload` after the anonymous collection is added, but before creating the comment. The issue is effectively that you've got multiple copies of the work in memory, and one of them is anonymous and the other isn't. (This doesn't happen in production because each request loads items from scratch, but tests sometimes need a `reload` call to make sure all objects have access to the same data.)